src/components/Search/ResultCard.tsx

```diff
@@ -28,13 +47,30 @@ export default function ResultCard({ data }: ResultCardProps) {
         <h2 className="margin-top-1 margin-bottom-3">{data.name}</h2>
       </Link>
       <BasicResultDetail result={data} isCondensed />
-      <Link
-        className="usa-button"
-        to={`/result/${data.id}`}
-        state={{ prevSearch: location.search, data }}
+      <Grid
+        row
+        mobile={{ col: true }}
+        className="flex-justify-center tablet:flex-justify flex-align-center"
       >
-        {t("fullDetail")}
-      </Link>
+        <Link
+          className="usa-button"
+          to={`/result/${data.id}`}
+          state={{ prevSearch: location.search, data }}
+        >
+          {t("fullDetail")}
+        </Link>
+        <Checkbox
+          id={`compare-${isMobile ? "mobile-" : ""}${data.id}`}
+          name={`Compare ${data.name}`}
+          label={t("compareCheckbox")}
+          checked={!!providers.find((provider) => provider.id === data.id)}
```

Comment:
nit: instead of doing this same computation multiple times, can set `const isSelectedForCompare = !!providers.find((provider) => provider.id === data.id)` and then it'll also be slightly easier to read